src/lib/sql/sql_translator.cpp

```diff
@@ -344,6 +347,51 @@ std::shared_ptr<AbstractLQPNode> SQLTranslator::_translate_select(const hsql::Se
   return current_result_node;
 }
 
+auto SQLTranslator::_classify_join_conditions(const hsql::JoinDefinition& join,
+                                              std::function<LQPChildSide(const hsql::Expr*)> get_side) {
+  std::vector<const hsql::Expr*> conditions;
+  _collect_join_conditions(join.condition, conditions);
+
+  std::vector<const hsql::Expr*> joins;
+  std::vector<const hsql::Expr*> left;
+  std::vector<const hsql::Expr*> right;
+
+  for (auto condition : conditions) {
+    const bool has_left_column = condition->expr && condition->expr->type == hsql::kExprColumnRef;
+    const bool has_right_column = condition->expr2 && condition->expr2->type == hsql::kExprColumnRef;
```

Comment:
Maybe Assert that one of them is true. Or just support it. You never know what kind of join conditions generated queries might contain

Reply:
> Or just support it.

Actually that is what caused this monster ;) Maybe the flag should be named `left_is_colum` or similar. Currently all possible combinations of columns with literals inside a a predicate are supported.